express/code/blocks/print-product-detail/utilities/data-formatting.js

```diff
@@ -158,8 +158,13 @@ export async function updateDataObjectProductDetails(dataObject, productDetails)
 }
 
 export function updateDataObjectProductPrice(dataObject, productPrice, quantity) {
-  const applicableDiscount = productPrice
-    ?.discountProductItems[1] || productPrice?.discountProductItems[0];
+  let applicableDiscount = null;
+  if (productPrice?.discountProductItems?.length > 0) {
+    applicableDiscount = productPrice.discountProductItems.reduce(
+      (max, item) => (item.discountPercent > max.discountPercent ? item : max),
```

Comment:
Looking good overall, should we add Number validation and default to 0 in case it is undefined or null?